cpssEnabler/linuxNoKernelModule/drivers/dmaDriver.c

```diff
@@ -28,320 +28,929 @@ disclaimer.
+		 (unsigned long long)m->phys, (unsigned long long)vma->vm_start,
+		 (unsigned long long)vma->vm_pgoff, (unsigned long)m->size);
+
+	if (remap_pfn_range(vma, vma->vm_start, vma->vm_pgoff, m->size,
+			    vma->vm_page_prot)) {
+		dev_err(m->dev, "remap_pfn_range failed\n");
+		return -ENXIO;
+	}
+#else /* MMAP_USE_REMAP_PFN_RANGE */
+	dev_info(m->dev, "dma_mmap_coherent(vma=0x%llx, c=%p, dma=0x%llx, size=0x%llx)\n",
+		(unsigned long long)vma->vm_start, m->virt,
+		(unsigned long long)m->dma, (unsigned long long)m->size);
+	if (dma_mmap_coherent(m->dev, vma, m->virt, m->dma, m->size)) {
+		dev_info(m->dev, "dma_mmap_coherent() failed\n");
+		return -ENXIO;
+	}
+#endif /* MMAP_USE_REMAP_PFN_RANGE */
+
+	return 0;
+}
 
-#if 0
-    /* disable caching on mapped memory */
-    vma->vm_page_prot = pgprot_noncached(vma->vm_page_prot);
-#endif
-    // TODO: check if write combine support is  needed ?
-    //vma->vm_page_prot = pgprot_writecombine(vma->vm_page_prot);
-
-    printk("remap_pfn_range(phys=0x%llx vm_start=0x%llx, vm_pgoff=0x%llx, vm_size=0x%lx, )\n",
-                (unsigned long long)m->phys,
-                (unsigned long long)vma->vm_start,
-                (unsigned long long)vma->vm_pgoff,
-                (unsigned long)m->size);
-
-    if (remap_pfn_range(vma, vma->vm_start, vma->vm_pgoff, m->size, vma->vm_page_prot)) {
-        printk("remap_pfn_range failed\n");
-        return -ENXIO;
-    }
-#else
-    printk("dma_mmap_coherent(dev=%s, vma=0x%llx, c=%p, dma=0x%llx, size=0x%llx)\n",
-                ((m->dev)?m->dev->kobj.name:"NULL"),
-                (unsigned long long)vma->vm_start,
-                m->virt,
-                (unsigned long long)m->dma,
-                (unsigned long long)m->size);
-    if (dma_mmap_coherent(m->dev, vma, m->virt, m->dma, m->size)) {
-        printk("dma_mmap_coherent() failed\n");
-        return -ENXIO;
-    }
-#endif
+static ssize_t mvDmaDrv_read(struct file *f, char *buf, size_t siz, loff_t *off)
+{
+	struct dma_mapping *m = (struct dma_mapping *)f->private_data;
+	unsigned long long dma;
 
-    return 0;
-}
+	if (!m)
+		return -EFAULT;
+
+	if (siz < sizeof(dma))
+		return -EINVAL;
+
+	dma = (unsigned long long)m->dma;
 
+	if (copy_to_user(buf, &dma, sizeof(dma)))
+		return -EFAULT;
 
-static loff_t mvDmaDrv_lseek(struct file * file, loff_t off, int unused)
+	return sizeof(dma);
+}
+
+static int mvDmaDrv_FreeDummys(struct dma_mapping *m)
 {
-    struct dma_mapping *m = (struct dma_mapping *)file->private_data;
-    struct pci_dev *pdev;
-    int domain = (off >> 16) & 0xffff;
-    unsigned int bus = (off >> 8) & 0xff;
-    unsigned int devfn = PCI_DEVFN(((off >> 3) & 0x1f), (off & 0x07));
-
-    if (!m)
-        return -EFAULT;
-    pdev = pci_get_domain_bus_and_slot(domain, bus, devfn);
-    if (pdev == NULL)
-        return -ENOENT;
-
-    m->dev = &(pdev->dev);
-    printk("selected PCI device %s d=%p\n", m->dev->kobj.name,m->dev);
-    return 0;
+	int i;
+
+/*	for (i=0; i<16; i++) {
```

Comment:
Commented out. Do we really need this function?

Reply:
Fixed